### PR TITLE
Refactored GameObject, added wheels

### PR DESCRIPTION
**Pull request by josepharhar**
This does not support animations, it is mostly for changing gameobjects to make them support a tree of models

### CODE REVIEW

**Comment by josepharhar** [comment]
will rebase when i get back from class at 8pm

**Comment by Matthew-Stewart** [comment]
QA 🏖

**Comment by josepharhar** [comment]
After i rebase this id like someone to QA the json storage of game objects, i might have fucked with it

File: src/game_renderer/GameRenderer.cpp [hunk 1]
```diff
@@ -384,13 +405,12 @@ void GameRenderer::RenderIt(GLFWwindow* window,
...
   current_texture = textures["nightsky"];
   current_texture->bind(current_program->getUniform("Texture0"));
-  MV = sky->GetTransform();
   glUniformMatrix4fv(current_program->getUniform("P"), 1, GL_FALSE,
                      glm::value_ptr(P->topMatrix()));
   glUniformMatrix4fv(current_program->getUniform("V"), 1, GL_FALSE,
                      glm::value_ptr(V.topMatrix()));
   glUniformMatrix4fv(current_program->getUniform("MV"), 1, GL_FALSE,
-                     glm::value_ptr(MV.topMatrix()));
+    glm::value_ptr(sky->GetTransform()));
```
Comment (bbeck13) [inline src/game_renderer/GameRenderer.cpp]: haha thanks we've definitely got some silly Zoe code lying around

**Comment by josepharhar** [comment]
Just finished rebasing, please CR/QA

**Review by bbeck13 — COMMENTED** [review]
I can't see the sky! Looks pretty good otherwise

**Comment by bbeck13** [comment]
I see the sky now
CR 👍 QA 🏡

**Pull request merged**